ReactiveCocoa/Swift/DynamicProperty.swift

```diff
@@ -1,29 +1,38 @@
 import Foundation
 import enum Result.NoError
 
+/// Models types that can be represented in Objective-C (i.e., reference
+/// types, including generic types when boxed via `AnyObject`).
+private protocol ObjectiveCRepresenting {
+	associatedtype Value
+	func extractValue(fromRepresentation representation: AnyObject) -> Value
+	func represent(value value: Value) -> AnyObject
+}
+
 /// Wraps a `dynamic` property, or one defined in Objective-C, using Key-Value
 /// Coding and Key-Value Observing.
 ///
 /// Use this class only as a last resort! `MutableProperty` is generally better
 /// unless KVC/KVO is required by the API you're using (for example,
 /// `NSOperation`).
-@objc public final class DynamicProperty: RACDynamicPropertySuperclass, MutablePropertyType {
-	public typealias Value = AnyObject?
+public final class DynamicProperty<Object>: MutablePropertyType {
```

Comment:
`Object` implies reference semantics, but this type isn't constrained to anything. Maybe `Value` would be better? Then the `Value` `typealias` is probably not needed.